src/modules/_nomodule/PixelManipulation.js

```diff
@@ -93,5 +94,10 @@ module.exports = function PixelManipulation(image, options) {
         options.output(options.image, datauri, options.format);
       if (options.callback) options.callback();
     });
+    }
+    if(res){
+      pixels=res;
+      generateOutput();
+    }
```

Comment:
Do we need to do this ??? as this just adds extra code, I understand that you want the pixels to be modified is there is a change but I guess this would also affect other modules.

Reply:
Actually it adds some code, but if you go through the above code where I have used the extraManipulation function using res as the returned output it gives us the flexibility to provide not only pixels but also other things as returned output.
Like in my case as QR.js required some callback as the code was async there it becomes easy this way and like this other modules added in future also do have the flexibility to use the same .
since this is nothing but a modular and DRY approach, it is easily adoptable without breaking any existing workflow that it should not.